src/escd/network/peerclientmanager.cpp

```diff
@@ -266,92 +277,81 @@ void PeerConnectManager::connect(std::string peer_address, unsigned short port,
+            }
+        }
     }
 }
 
-void PeerConnectManager::connectPeersFromDNS(int& connNeeded)
+void PeerConnectManager::getPeersFromDNS(std::vector<std::pair<in_addr_t, unsigned short>> &peerAddrs)
 {
     try
     {
         boost::asio::ip::tcp::resolver              resolver(m_ioService);
         boost::asio::ip::tcp::resolver::query       query(m_opts.dnsa.c_str(),SERVER_PORT);
         boost::asio::ip::tcp::resolver::iterator    iterator = resolver.resolve(query);
         boost::asio::ip::tcp::resolver::iterator    end;
-        std::map<uint32_t,boost::asio::ip::tcp::resolver::iterator> endpoints;
 
         while(iterator != end)
         {
-            uint32_t r=random()%0xFFFFFFFF;
-            endpoints[r]=iterator++;
-        }
+            auto addr = iterator->endpoint().address().to_string();
+            auto port = iterator->endpoint().port();
 
-        for(auto ep=endpoints.begin(); ep!=endpoints.end(); ep++)
-        {
-            if(connNeeded <= 0){
-                break;
+            in_addr   inaddr;
+            if(inet_aton(addr.c_str(), &inaddr) != 0){
+                peerAddrs.push_back(std::make_pair(inaddr.s_addr, port));
             }
-
-            connect(ep->second->endpoint(), BANK_MAX);
-            --connNeeded;
+            ++iterator;
         }
     }
     catch (std::exception& e) {
         ELOG("DNS Connect Exception: %s\n", e.what());
     }
 }
 
-void PeerConnectManager::connectPeersFromServerFile(int& connNeeded)
+void PeerConnectManager::getPeersFromServerFile(std::vector<std::pair<in_addr_t, unsigned short>> &peerAddrs)
 {
-    while(connNeeded>0)
-    {
-        bool        foundNew{false};
-        node        nodeInfo;
-        uint16_t    nodeIndx      = m_server.getRandomNodeIndx();
-        uint16_t    maxNodeIndx   = m_server.getMaxNodeIndx();
-
-        if(maxNodeIndx == 0){
-            assert(maxNodeIndx != 0);
-            ELOG("maxNodeIndx == 0, returning\n");
-            return;
-        }
+    node        nodeInfo;
+    uint16_t    maxNodeIndx   = m_server.getMaxNodeIndx();
+
+    if(maxNodeIndx == 0){
+        assert(maxNodeIndx != 0);
+        ELOG("maxNodeIndx == 0, returning\n");
+        return;
+    }
 
-        //@TODO: improve efficiency of finding new node
-        for(int i = 0; i<=maxNodeIndx; ++i)
+    //@TODO: improve efficiency of finding new node
+    for(int i = 0; i<=maxNodeIndx; ++i)
+    {
+        //Don't connect to myself
+        if(i != m_opts.svid && m_server.getNode(i, nodeInfo))
         {
-            //Don't connect to myself
-            if(nodeIndx != m_opts.svid && m_server.getNode(nodeIndx, nodeInfo))
-            {
-                if(nodeInfo.port > 0
-                        && m_peers.find(std::make_pair(nodeInfo.ipv4, nodeInfo.port)) == m_peers.end() )
-                {
-                    foundNew = true;
-                    break;
-                }
+            if(nodeInfo.mtim >= m_server.last_srvs_.now - BLOCKSEC*BLOCKDIV) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)